### ISSUE TITLE
SeqIO.write() documentation suggestion

### ISSUE THREAD

**Issue by singing-scientist** [biopython/biopython#3854]
### Setup

Biopython version=1.79
Python version=3.8.8
OS=macOS Monterey v12.2

### Suggestion

I discovered through trial and error that SeqIO.write() appends a record (rather than overwrites the output file) if provided with an already-open filehandle. This was an incredibly useful discovery, as there is no mention of appending in the documentation, and I had assumed the overwriting behavior described in the [Tutorial](http://biopython.org/DIST/docs/tutorial/Tutorial.html) was the only option. I therefore thought I needed to build a list of recs, which could only later be written all at once (a potentially memory-prohibitive approach). I am new to both Python and Biopython, so this could well be a careless oversight on my part. However, perhaps it would be worth describing this behavior more explicitly (i.e., using the word 'append') in the help(SeqIO.write) documentation?

Thanks for this amazing module!

**Comment by peterjc** [comment]
Yes you can, but only for file formats which write the records sequentially with no special header/footer. So for example, this works fine for FASTA, FASTQ, GenBank, EMBL - but fails on SFF.

Could you suggest a wording change to the help text (docstrings) in https://github.com/biopython/biopython/blob/master/Bio/SeqIO/__init__.py and/or the Tutorial, probably this chapter: https://github.com/biopython/biopython/blob/master/Doc/Tutorial/chapter_seqio.tex

**Comment by singing-scientist** [comment]
Thanks so much for this clarification -- this makes perfect sense! I just forked/committed suggestions to both files, but pasting below in case that didn't work:

1. in write() docstring:

```
For handle, if an already-existing filename (string) is provided, it will
    be overwritten; if an open file handle is provided, the record will be
    appended to the file, which may result in valid (e.g., FASTA) or invalid
    (e.g., SFF) output depending on the file format. Note if providing a file
    handle, your code should close the handle after calling this function (to
    ensure the data gets flushed to disk).
```

2. in Tutorial:

```
\emph{Note} - If you tell the \verb|Bio.SeqIO.write()| function to write to a file that already exists, the old file will be overwritten without any warning. On the other hand, if you tell the \verb|Bio.SeqIO.write()| function to write to an open file handle, the record will be appended to the file, which may result in valid (e.g., FASTA) or invalid (e.g., SFF) output depending on the file format.
```

**Comment by mdehoon** [comment]
> Note if providing a file
    handle, your code should close the handle after calling this function (to
    ensure the data gets flushed to disk).

Can you reword this? If the user wants to call `SeqIO.write` repeatedly with the same open file handle, they should not close the handle. Also, the data can be written to disk without closing the handle by calling `flush`.

**Comment by singing-scientist** [comment]
Thanks for the great improvement! How's the following?

> For handle, if an already-existing filename (string) is provided, it will
>     be overwritten; if an open file handle is provided, the record will be
>     appended to the file, which may result in valid (e.g., FASTA) or invalid
>     (e.g., SFF) output depending on the file format. If the user wants to call
>     SeqIO.write repeatedly with the same open file handle, they should not
>     close the handle until all records have been written. To ensure all data
>     are written to disk, the user may call flush or close the handle.

**Comment by hubutui** [comment]
Hi, guys, I come across this issue when I'm trying to read a large fasta file which could not be loaded to memory all at once, translate it from DNA to protein, and save it to fasta. I would be great if we could write something like:
```python
Bio.SeqIO.write(sequences, handle, format, mode='append')
```

**Comment by hubutui** [comment]
@peterjc Yeah, thanks. I just check out and make sure for fasta format, multiple write to the same handle when append sequence to the outputfile.
